lua/entities/gmod_tardis/modules/sh_hads.lua

```diff
@@ -34,6 +34,19 @@ if SERVER then
             self:AutoDemat()
             self:CallHook("HADSTrigger")
         end
+    end 
+
+    ENT:AddHook("OnTakeDamage", "hads", function(self)
+        self:TryTriggerHADS()  
+    end)
+
+    hook.Add("OnPhysgunPickup", "tardis-hads", function(ply,ent)
+        if ent:GetClass()=="gmod_tardis" then
+            ent:TryTriggerHADS()
```

Comment:
As above when TriggerHADS returns true/false we can use that in the if check below rather than checking hads-triggered here